### PR TITLE
Improve posthog

### PR DESCRIPTION
**Pull request by arcticfly**
## Changes
* Identify posthog users
* Rewrite posthog events

We should add `NEXT_PUBLIC_HOST` to our production server as well.

### CODE REVIEW

**Review by corbt — APPROVED** [review]

File: src/utils/analytics.ts [hunk 1]
```diff
@@ -1,13 +1,21 @@
 // Make sure we're in the browser
```
Comment (corbt) [inline src/utils/analytics.ts]: Can we just delete this file and move it all to SessionIdentifier?

File: src/utils/analytics.ts [hunk 2]
```diff
@@ -1,13 +1,21 @@
...
     posthog.init(env.NEXT_PUBLIC_POSTHOG_KEY, {
-      api_host: "https://app.posthog.com",
+      api_host: `${env.NEXT_PUBLIC_HOST}/ingest`,
     });
   }
 }
+
+export const identifySession = (session: Session) => {
+  if (!enableAnalytics) return;
+  posthog.identify(session.user?.id, {
```
Comment (corbt) [inline src/utils/analytics.ts]: What happens if `user` is null? Should we check for that and not call `identify` in that case?

Comment (arcticfly) [reply src/utils/analytics.ts]: Makes sense

File: src/utils/analytics.ts (hunk 1 above)
Comment (arcticfly) [reply src/utils/analytics.ts]: I'll delete SessionIdentifier and move it in here, since we'll likely add more helper functions later.

**Pull request merged**